packages/go_router/example/lib/async_redirection.dart

```diff
@@ -0,0 +1,244 @@
+  Widget build(BuildContext context) => MaterialApp.router(
+        routeInformationProvider: _router.routeInformationProvider,
+        routeInformationParser: _router.routeInformationParser,
+        routerDelegate: _router.routerDelegate,
+        title: title,
+        debugShowCheckedModeBanner: false,
+      );
+
+  late final GoRouter _router = GoRouter(
+    routes: <GoRoute>[
+      GoRoute(
+        path: '/',
+        builder: (BuildContext context, GoRouterState state) =>
+            const HomeScreen(),
+      ),
+      GoRoute(
+        path: '/login',
+        builder: (BuildContext context, GoRouterState state) =>
+            const LoginScreen(),
+      ),
+    ],
+
+    // redirect to the login page if the user is not logged in
+    redirect: (BuildContext context, GoRouterState state) async {
+      // Using `of` method creates a dependency of StreamAuthScope. It will
+      // cause go_router to reparse current route if StreamAuth has new sign-in
+      // information.
+      final bool loggedIn = await StreamAuthScope.of(context).isSignedIn();
+      final bool loggingIn = state.subloc == '/login';
+      if (!loggedIn) {
+        return loggingIn ? null : '/login';
+      }
+
+      // if the user is logged in but still on the login page, send them to
+      // the home page
+      if (loggingIn) {
+        return '/';
+      }
+
+      // no need to redirect at all
+      return null;
+    },
+  );
+}
+
+/// The login screen.
+class LoginScreen extends StatefulWidget {
+  /// Creates a [LoginScreen].
+  const LoginScreen({Key? key}) : super(key: key);
+
+  @override
+  _LoginScreenState createState() => _LoginScreenState();
+}
+
+class _LoginScreenState extends State<LoginScreen>
+    with TickerProviderStateMixin {
+  bool loggingIn = false;
+  late final AnimationController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
+      vsync: this,
+      duration: const Duration(seconds: 1),
+    )..addListener(() {
+        setState(() {});
+      });
+    controller.repeat();
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) => Scaffold(
+        appBar: AppBar(title: const Text(App.title)),
+        body: Center(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              if (loggingIn) CircularProgressIndicator(value: controller.value),
+              if (!loggingIn)
+                ElevatedButton(
+                  onPressed: () {
+                    // log a user in, letting all the listeners know
+                    StreamAuthScope.of(context).signIn('test-user');
+                    setState(() {
+                      loggingIn = true;
+                    });
+                    // router will automatically redirect from /login to / using
+                    // refreshListenable
```

Comment:
I'd move that redirect comment up. Also, I'd capitalize all comments and add punctuation to all comments in this file for consistency.

```suggestion
                    // Log a user in and notify all listeners.
                    // This will automatically redirect from /login to / using
                    // refreshListenable.
                    StreamAuthScope.of(context).signIn('test-user');
                    setState(() {
                      loggingIn = true;
                    });
```

Reply:
oops it was a outdated comment i forgot to delete, it no longer apply